File: pocketmine/src/Taylcd/TextReplacer.php
<?php

namespace Taylcd;

use pocketmine\event\Listener;
use pocketmine\event\server\DataPacketSendEvent;
use pocketmine\network\protocol\TextPacket;
use pocketmine\plugin\PluginBase;

class TextReplacer extends PluginBase implements Listener
{
    public $list = [];

    public function onEnable()
    {
        $this->saveDefaultConfig();
        $this->reloadConfig();

        $this->list = $this->getConfig()->getAll();
        $this->getServer()->getPluginManager()->registerEvents($this, $this);
    }

    public function onDataPacketSend(DataPacketSendEvent $event)
    {
        $pk = $event->getPacket();
        if(!$pk instanceof TextPacket) return;
        $name = $event->getPlayer()->getName();
        foreach($this->list as $item=>$value)
        {
            $item = str_replace('{PLAYER}', $name, $item);
            $value = str_replace('{PLAYER}', $name, $value);

            $pk->message = str_replace($item, $value, $pk->message);
        }
    }
}

File: nukkit/src/taylcd/TextReplacer.java
package taylcd;

import cn.nukkit.event.EventHandler;
import cn.nukkit.event.EventPriority;
import cn.nukkit.event.Listener;
import cn.nukkit.event.server.DataPacketSendEvent;
import cn.nukkit.network.protocol.DataPacket;
import cn.nukkit.network.protocol.TextPacket;
import cn.nukkit.plugin.PluginBase;

import java.io.File;
import java.util.Calendar;
import java.util.HashMap;
import java.util.Map;

public class TextReplacer extends PluginBase implements Listener {

    private Map<String, Object> list;

    @Override
    public void onLoad() {

        File file = new File(getDataFolder(), "");
        if(!file.isDirectory()) if(!file.mkdir()) getLogger().notice("Unable to create data folder.");

        this.saveDefaultConfig();
        this.reloadConfig();

        this.list = this.getConfig().getAll();

    }

    @Override
    public void onEnable() {

        this.getServer().getPluginManager().registerEvents(this, this);

    }

    @EventHandler(priority = EventPriority.LOW, ignoreCancelled = true)
    public void onDataPacketSend(DataPacketSendEvent event) {

        DataPacket pk = event.getPacket();
        if(pk instanceof TextPacket) {

            String name = event.getPlayer().getName();
            Calendar time = Calendar.getInstance();

            HashMap<String, String> variables = new HashMap<String, String>() {
                {
                    put("{PLAYER}", name);
		            put("{LEVEL}", event.getPlayer().getLevel().getName());
                }
            };

            for(Map.Entry<String, Object> entry : this.list.entrySet()) {

                String search = entry.getKey();
                String replacement = entry.getValue().toString();

                for(Map.Entry<String, String> variable : variables.entrySet()) {

                    search = search.replace(variable.getKey(), variable.getValue());
                    replacement = replacement.replace(variable.getKey(), variable.getValue());

                }

                ((TextPacket) pk).message = ((TextPacket) pk).message.replace(search, replacement);

            }

        }

    }

}
